app/concerns/dry_runnable.rb

```diff
@@ -14,7 +14,12 @@ class << self
 
     begin
       raise "#{short_type} does not support dry-run" unless can_dry_run?
-      check
+      if event
+        raise "This agent cannot receive an event!" unless can_receive_events?
+        receive([event])
```

Comment:
Some agents can both `check` and `receive` events, like the website agent.  I guess specifying the event in the UI is how you select which mode to dry run?